connect/src/com/telenor/connect/AbstractSdkProfile.java

```diff
@@ -166,6 +166,10 @@ protected void previewParameters(ParametersHolder parameters) {
                         && isEmpty(wellKnownConfig.getNetworkAuthenticationTargetUrls()))) {
                     parameters.add("login_hint", String.format("MCCMNC:%s", mccMnc));
         }
+
+        if (!ConnectSdk.isCellularDataNetworkConnected()) {
```

Comment:
Ok, I think the problem comes from 
networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE);
It does not work basically.
If the method is  
public static boolean isCellularDataNetworkConnected() {
  networkInfo = connectivityManager.getNetworkInfo(cellularNetwork);
  return (networkInfo != null) && networkInfo.isConnected();
}
it seems to work OK.